### PR TITLE
Add Visual Studio Code Insiders build for beta stream

### PR DESCRIPTION
**Pull request by fooishbar**
Based on the existing com.visualstudio.code build, containing the
release stream of VS Code, add a separate app-ID for the Insiders
stream, which is their beta release channel.

Disclaimers have been added to the app description and warning file
opened at first startup, however this should still only be distributed
in a beta channel, and not generally.

### CODE REVIEW

File: com.visualstudio.code.insiders.appdata.xml [hunk 1]
```diff
@@ -0,0 +1,27 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (nedrichards) [inline com.visualstudio.code.insiders.appdata.xml]: huh, this file should have a launchable tag

https://github.com/flathub/flathub/wiki/AppData-Guidelines

`<launchable type="desktop-id">com.visualstudio.code.insiders.desktop</launchable>`

File: com.visualstudio.code.insiders.yaml [hunk 2]
```diff
@@ -0,0 +1,93 @@
+app-id: com.visualstudio.code.insiders
+default-branch: stable
+base: org.electronjs.Electron2.BaseApp
+base-version: '19.08'
+runtime: org.freedesktop.Sdk
+runtime-version: '19.08'
+sdk: org.freedesktop.Sdk
+command: code-insiders
+tags: [proprietary]
+desktop-file-name-suffix: ' (Flatpak)'
```
Comment (nedrichards) [inline com.visualstudio.code.insiders.yaml]: So this is going to say `Visual Studio Code (Insiders) (Flatpak)` right? Seems a bit ungainly

File: flathub.json [hunk 3]
```diff
@@ -0,0 +1,4 @@
+{
+  "only-arches": ["x86_64"],
+  "skip-appstream-check": true
```
Comment (nedrichards) [inline flathub.json]: hah! we shouldn't have this in here any more ideally.

**Comment by nedrichards** [comment]
Note: previous discussion on https://github.com/flathub/com.visualstudio.code/pull/147

**Comment by nedrichards** [comment]
bot, build com.visualstudio.code.insiders

File: com.visualstudio.code.insiders.yaml [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+          - mv usr/share/code-insiders vscode-insiders
+          - rm -r code-insiders.deb control.tar.gz data.tar.xz debian-binary usr
+      - type: file
+        path: code-insiders.sh
+      - type: file
+        path: flatpak-warning.txt
+      - type: file
+        path: npmrc
+      - type: file
+        path: com.visualstudio.code.insiders.appdata.xml
```
Comment (LeandroStanger) [inline com.visualstudio.code.insiders.yaml]: New appdata files should use metainfo instead. https://www.freedesktop.org/software/appstream/docs/chap-Quickstart.html

**Comment by eszlari** [comment]
I guess the advantage of creating a new repo and not just a beta branch for the existing, is that when having both, stable and insiders installed, configs are not overwritten:

https://github.com/flatpak/flatpak/issues/2746

**Comment by nedrichards** [comment]
@eszlari right - Insiders is specifically designed to be run side by side. (see the discussion in the main version PR I linked to above)

**Comment by eszlari** [comment]
> @eszlari right - Insiders is specifically designed to be run side by side. (see the discussion in the main version PR I linked to above)

Yeah, I just checked this myself. So what's the point in creating a new repo?

**Comment by nedrichards** [comment]
You need a new app-id to get full side by side. The build system binds the app that is published to the app-id of the repo etc. so you can't just chuck it into the beta channel on the same repo and you can only have one branch of an app exported at the same time (see `flatpak make-current`).

**Comment by nedrichards** [comment]
I performed most of the suggested changes on the production version here just now https://github.com/flathub/com.visualstudio.code/commit/a81f21309d477a59e107de66f5dc6da659bac96c

**Comment by fooishbar** [comment]
@nedrichards Thanks, I pulled in your changes from the main repo.

**Comment by nedrichards** [comment]
bot, build com.visualstudio.code.insiders

**Comment by nedrichards** [comment]
In testing the contents of the flatpak-warning.txt were not correctly displayed.

**Comment by fooishbar** [comment]
@nedrichards Oops. I've fixed that, revved the version to the newest available, and also actually removed `skip-appstream-check` this time.

**Comment by LeandroStanger** [comment]
bot, build com.visualstudio.code.insiders

**Pull request closed without merging**